admin/initFiles/conf/settings.md
# Settings JSON Explanation

This document describes the structure and purpose of the `settings.json` configuration file.

## Overview

The `settings.json` file contains various configuration parameters used by the application. It includes settings for security, debugging, localization, date/time formatting, and input formats.

## JSON Structure

```json
{
  "secret": null,
  "debug": false,
  "charset": "utf-8",
  "language": "pt_br",
  "time_zone": "America/Sao_Paulo",
  "tz": true,
  "formats": {
    "date": "N j, Y",
    "datetime": "N j, Y, P",
    "time": "P",
    "year_month": "F Y",
    "month_day": "F j",
    "short_date": "d/m/Y",
    "short_datetime": "d/m/Y P",
    "number": {
      "separator": {
        "decimal": ".",
        "thousand": ",",
        "group": 0
      }
    },
    "week": {
      "first_day": 0
    },
    "input": {
      "date": {
        "%Y-%m-%d": "2000-10-25",
        "%d/%m/%Y": "25/10/2000",
        "%m/%d/%Y": "10/25/2000",
        "%d/%m/%y": "25/10/00",
        "%m/%d/%y": "10/25/00",
        "%d %b %Y": "25 Oct 2000",
        "%d %b, %Y": "25 Oct, 2000",
        "%b %d %Y": "Oct 25 2000",
        "%b %d, %Y": "Oct 25, 2000",
        "%d %B %Y": "25 October 2000",
        "%d %B, %Y": "25 October, 2000",
        "%B %d %Y": "October 25 2000",
        "%B %d, %Y": "October 25, 2000"
      },
      "time": {
        "%H:%M:%S": "14:30:59",
        "%H:%M:%S.%f": "14:30:59.001200",
        "%H:%M": "14:30"
      }
    }
  },
	"log": {
		"path": "project/logs",
		"file": "system.log",
		"createPath": false,
		"maxSize": 1048576,
		"level": {
			"verbose": 16,
			"file": 0
		}
	}
}
```

## Key Descriptions

- `name`: Name of project. It is currently set to `null`.
- `secret`: A secret key used for cryptographic operations. It is currently set to `null`.
- `debug`: A boolean value indicating whether the application is in debug mode. Set to `false` for production environments.
- `charset`: The default character encoding used by the application. Set to `utf-8`.
- `language`: The language code for localization. Set to `pt_br` (Brazilian Portuguese).
- `time_zone`: The time zone for the application. Set to `America/Sao_Paulo`.
- `tz`: A boolean value to enable or disable the timezone display. Set to `true`.
- `formats`: An object containing formatting settings for dates, times, and numbers. <https://docs.python.org/library/datetime.html#strftime-behavior>
    - `date`: The format string for dates.
    - `datetime`: The format string for date and time.
    - `time`: The format string for times.
    - `year_month`: The format string for year and month.
    - `month_day`: The format string for month and day.
    - `short_date`: A shorter date format.
    - `short_datetime`: A shorter date and time format.
    - `number`:
      - `separator`:
        - `decimal`: The decimal separator.
        - `thousand`: The thousands separator.
        - `group`: The grouping size.
    - `week`:
      - `first_day`: The first day of the week (0 for Sunday).
    - `input`:
      - `date`: Various date format strings and example values.
      - `time`: Various time format strings and example values.
  - `log`: Log config
    - `path`: Folder of logs
    - `file`: File name of the system log
    - `createPath`: If True and if there are not, it will create path automatily
    - `maxSize`: max size file log
    - `level`: Levels see LOG constants
      - `verbose`: Log screen
      - `file`: Log File
## Usage

This configuration file should be used to customize the application's behavior according to the specific deployment environment and localization requirements.


conf/settings.py
class Settings:
    secret = None
    """Secret key to crypt sensible data"""

    debug = False
    """True only to development"""

    charset = "utf-8"
    """Default charset to use"""

    language = "pt_br"
    """Language of system <http://www.i18nguy.com/unicode/language-identifiers.html>"""

    time_zone = "America/Chicago"
    """Local time zone <https://en.wikipedia.org/wiki/List_of_tz_zones_by_name>"""

    tz = True
    """Show or not timezone"""

    formats = {
        "date": "N j, Y",
        "datetime": "N j, Y, P",
        "time": "P",
        "year_month": "F Y",
        "month_day": "F j",
        "short_date": "m/d/Y",
        "short_datetime": "m/d/Y P",
        "number": {
            "separator": {
                "decimal": ".",
                "thousand": ",",
                "group": 0
            }
        },
        "week": {
            "first_day": 0
        },
        "input": {
            "date": {
                "%Y-%m-%d": "2006-10-25",
                "%m/%d/%Y": "10/25/2006",
                "%m/%d/%y": "10/25/06",
                "%b %d %Y": "Oct 25 2006",
                "%b %d, %Y": "Oct 25, 2006",
                "%d %b %Y": "25 Oct 2006",
                "%d %b, %Y": "25 Oct, 2006",
                "%B %d %Y": "October 25 2006",
                "%B %d, %Y": "October 25, 2006",
                "%d %B %Y": "25 October 2006",
                "%d %B, %Y": "25 October, 2006"
            },
            "time": {
                "%H:%M:%S": "14:30:59",
                "%H:%M:%S.%f": "14:30:59.000200",
                "%H:%M": "14:30"
            }
        }
    }
    """<https://docs.djangoproject.com/en/dev/ref/templates/builtins/#date> <https://docs.python.org/library/datetime.html#strftime-behavior>"""

    contact: list[dict] = []
    """
    List of contacts of project.

    Returns:
        settings.json (part)
        ```json
        "contact": [
            {
                "name": "Full name",
                "e_mail": [
                    "email@example.com"
                ],
                "phones": [
                    "+55 31 99999-1234"
                ]
            }
        ]
        ```
    """


    """

    # SETTINGS Documentation


    - Cache to store session data if using the cache session backend

    SESSION_CACHE_ALIAS = "default"

    - Cookie name. This can be whatever you want

    SESSION_COOKIE_NAME = "sessionid"

    - Age of cookie, in seconds (default: 2 weeks)

    SESSION_COOKIE_AGE = 60 *60* 24 *7* 2

    - A string like "example.com", or None for standard domain cookie

    SESSION_COOKIE_DOMAIN = None

    - Whether the session cookie should be secure (https:// only)

    SESSION_COOKIE_SECURE = false

    - The path of the session cookie

    SESSION_COOKIE_PATH = "/"

    - Whether to use the HttpOnly flag

    SESSION_COOKIE_HTTPONLY = True

    - Whether to set the flag restricting cookie leaks on cross-site requests

    - This can be "Lax", "Strict", "None", or false to disable the flag

    SESSION_COOKIE_SAMESITE = "Lax"

    - Whether to save the session data on every request

    SESSION_SAVE_EVERY_REQUEST = false

    - Whether a user"s session cookie expires when the web browser is closed

    SESSION_EXPIRE_AT_BROWSER_CLOSE = false

    - The module to store session data

    SESSION_ENGINE = "django.contrib.sessions.backends.db"

    - Directory to store session files if using the file session module. If None

    - the backend will use a sensible default

    SESSION_FILE_PATH = None

    - class to serialize session data

    SESSION_SERIALIZER = "django.contrib.sessions.serializers.JSONSerializer"

    SECURE_CONTENT_TYPE_NOSNIFF = True
    SECURE_CROSS_ORIGIN_OPENER_POLICY = "same-origin"
    SECURE_HSTS_INCLUDE_SUBDOMAINS = false
    SECURE_HSTS_PRELOAD = false
    SECURE_HSTS_SECONDS = 0
    SECURE_REDIRECT_EXEMPT = []
    SECURE_REFERRER_POLICY = "same-origin"
    SECURE_SSL_HOST = None
    SECURE_SSL_REDIRECT = false
    """
